hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/ReadableFreezeStore.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.state;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+
+/**
+ * Provides read-only methods for interacting with the underlying data storage mechanisms for
+ * working with freeze state
+ */
+public interface ReadableFreezeStore {
```

Comment:
Is that the correct place for the interface? I would have expected it in the `hedera-networkadmin-service` module.

Reply:
We had a long discussion about this a few weeks ago and the conclusion was that it should go here. Reasoning is that unlike the other stores, it is a wrapper around state that is shared with platform. Implementation of the underlying data store class is in HederaApp, not in a module.